File: app/src/main/java/fr/driffaud/band2fit/SettingsActivity.java
package fr.driffaud.band2fit;

import android.os.Bundle;

public class SettingsActivity extends AppCompatPreferenceActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        getFragmentManager().beginTransaction().replace(android.R.id.content, new MainPreferenceFragment()).commit();
    }

    @Override
    protected boolean isValidFragment(String fragmentName) {
        return super.isValidFragment(fragmentName);
    }
}

File: app/src/main/java/fr/driffaud/band2fit/SettingsActivity.kt
package fr.driffaud.band2fit

import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity
import java.io.File

class SettingsActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)

        // used by assets/logback.xml since the location cannot be statically determined
        val dir: File = applicationContext.getExternalFilesDirs(null)[0]
        System.setProperty("MI_EXPORTER_LOG_DIR", dir.absolutePath)

        supportFragmentManager
            .beginTransaction()
            .replace(android.R.id.content, PreferenceFragment())
            .commit()
    }

}
